presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -1388,6 +1517,14 @@ private static void throwDivisionByZeroException()
         throw new ArithmeticException("Division by zero");
     }
 
+    private static void multiplyShiftDestructive(Slice decimal, Slice multiplier, int rightShifts)
```

Comment:
Instead of these xxxDestructive methods, why not take an extra "result" argument and have the caller pass the same value as the source and output if it cares about reusing the object?